pkg/cmd/target/remove.go

```diff
@@ -50,16 +54,76 @@ var targetRemoveCmd = &cobra.Command{
 			selectedTargetName = args[0]
 		}
 
-		client, err := apiclient.GetApiClient(nil)
+		ctx := context.Background()
+		client, err := apiclient_util.GetApiClient(nil)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		res, err := client.TargetAPI.RemoveTarget(context.Background(), selectedTargetName).Execute()
+		if yesFlag {
+			fmt.Println("Deleting all workspaces.")
+			err := RemoveTargetWorkspaces(ctx, client, &selectedTargetName)
+
+			if err != nil {
+				log.Fatal(err)
+			}
+		} else {
+			form := huh.NewForm(
+				huh.NewGroup(
+					huh.NewConfirm().
+						Title(fmt.Sprintf("Delete all workspaces within %s?", selectedTargetName)).
+						Description("You might not be able to easily remove these workspaces later.").
+						Value(&yesFlag),
+				),
+			).WithTheme(views.GetCustomTheme())
+
+			err := form.Run()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			if yesFlag {
+				err := RemoveTargetWorkspaces(ctx, client, &selectedTargetName)
+				if err != nil {
+					log.Fatal(err)
+				}
+			} else {
+				fmt.Println("Proceeding with target removal without deleting workspaces.")
+			}
+		}
+
+		res, err := client.TargetAPI.RemoveTarget(ctx, selectedTargetName).Execute()
 		if err != nil {
-			log.Fatal(apiclient.HandleErrorResponse(res, err))
+			log.Fatal(apiclient_util.HandleErrorResponse(res, err))
 		}
 
 		views.RenderInfoMessageBold(fmt.Sprintf("Target %s removed successfully", selectedTargetName))
 	},
 }
+
+func init() {
+	targetRemoveCmd.Flags().BoolVarP(&yesFlag, "yes", "y", false, "Confirm deletion of all workspaces without prompt")
+}
+
+func RemoveTargetWorkspaces(ctx context.Context, client *apiclient.APIClient, target *string) error {
+	workspaceList, res, err := client.WorkspaceAPI.ListWorkspaces(ctx).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
+
+	for _, workspace := range workspaceList {
+		if workspace.Target != target {
```

Comment:
This check will always fail because you're comparing pointers. You need to compare values.